staging/src/k8s.io/kubectl/pkg/cmd/apply/apply.go

```diff
@@ -540,6 +544,64 @@ See https://kubernetes.io/docs/reference/using-api/server-side-apply/#conflicts`
 			return err
 		}
 
+		// Check if the returned object needs to have its
+		// kubectl-client-side-apply managed fields migrated server-side-apply.
+		//
+		// field ownership metadata is stored in three places:
+		// 	- server-side managed fields
+		//  - client-side managed fields
+		//  and the last_applied_configuration annotation.
+		//
+		// The migration merges the client-side-managed fields into the
+		// server-side-managed fields, leaving the last_applied_configuration
+		// annotation in place (since SSA will keep it up-to-date)
+		//
+		// By checking afterward instead of fetching the object beforehand and
+		// unconditionally fetching we can make 3 network requests in the rare
+		// case of migration and 1 request if migration is unnecessary.
+		//
+		// To check beforehand means 2 requests for most operations, and 3
+		//	requests in worst case.
+		if patchData, err := o.calculateSSAMigrationPatch(obj); err != nil {
+			// Print-error as a warning.
+			// This is a non-fatal error because object was successfully applied,
+			// but it might have issues since migration failed.
+			//
+			// This migration will be re-attempted if neccessary upon next
+			// apply.
+			fmt.Fprintf(o.ErrOut, warningMigrationPatchFailed, err.Error())
+		} else if patchData != nil {
+			// Send the patch to upgrade the managed fields if it is non-nil
+			_, err = helper.Patch(
+				info.Namespace,
+				info.Name,
+				types.JSONPatchType,
+				patchData,
+				nil,
+			)
+
+			if err != nil {
+				// Print as warning. Non-fatal since patch was already applied
+				fmt.Fprintf(o.ErrOut, warningMigrationPatchFailed, err.Error())
+			} else {
+				// Re-send original SSA patch (this will allow dropped fields to
+				// finally be removed)
+				obj, err = helper.Patch(
```

Comment:
Just to clarify: since `obj` is updated here, and later "Refreshed" in the object, I think the result of `kubectl apply --server-side -o yaml` will show the latest applied object, which is certainly what we want.